src/core_modules/capture-core/components/SearchBox/SearchBox.container.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+// $FlowFixMe
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import React, { useCallback } from 'react';
+import { useLocationQuery } from '../../utils/routing';
+import { SearchBoxComponent } from './SearchBox.component';
+import {
+    cleanSearchRelatedData,
+    navigateToNewTrackedEntityPage,
+    showInitialViewOnSearchBox,
+} from './SearchBox.actions';
+import { useCurrentTrackedEntityTypeId } from '../../hooks';
+import { ResultsPageSizeContext } from '../Pages/shared-contexts';
+import { usePreselectedProgram } from './hooks';
+
+export const SearchBox = () => {
+    const dispatch = useDispatch();
+    const { programId } = useLocationQuery();
+    const trackedEntityTypeId = useCurrentTrackedEntityTypeId();
+    const preselectedProgramId = usePreselectedProgram(programId);
+
+    const dispatchShowInitialSearchBox = useCallback(
+        () => { dispatch(showInitialViewOnSearchBox()); },
+        [dispatch]);
+    const dispatchCleanSearchRelatedData = useCallback(
+        () => { dispatch(cleanSearchRelatedData()); },
+        [dispatch]);
+    const dispatchNavigateToNewTrackedEntityPage = useCallback(() => {
+        dispatch(navigateToNewTrackedEntityPage());
+    }, [dispatch]);
+
+    const { searchStatus, error, ready, searchableFields, minAttributesRequiredToSearch } = useSelector(
+        ({ searchDomain, activePage }) => ({
+            searchStatus: searchDomain.searchStatus,
+            ready: !activePage.isLoading && !activePage.lockedSelectorLoads,
+            error: activePage.selectionsError && activePage.selectionsError.error,
```

Comment:
Can you look into the logic here. Are these needed? `activePage` is a concept we should get rid of. We should strive for the following flow: programId as input (and maybe others that I can't think of?) -> SearchBox components get the metadata they need internally based on the input. Feel free to use the Redux store for now (except `activePage`).

Reply:
Code updated @JoakimSM. Thanks for the feedback!